### ISSUE TITLE
DLQ input plugin fails to load

### ISSUE THREAD

**Issue by dedemorton** [elastic/logstash#7314]
Testing on 6.0.0-alpha2 (build candidate 2).

Here's the config:

```
input {
  dead_letter_queue {
    path => "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/data/dead_letter_queue"
    commit_offsets => false
    pipeline_id => "main"
  }
}

output {
  stdout {
    codec => rubydebug
  }
}
```
Version of the plugin that shows up when I list installed plugins:

```
logstash-input-dead_letter_queue (1.0.0)
```

Results in the following error:

```
[2017-06-05T09:44:22,412][ERROR][logstash.plugins.registry] Problems loading a plugin with {:type=>"input", :name=>"dead_letter_queue", :path=>"logstash/inputs/dead_letter_queue", :error_message=>"NameError", :error_class=>NameError, :error_backtrace=>["/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/plugins/registry.rb:225:in `namespace_lookup'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/plugins/registry.rb:161:in `legacy_lookup'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/plugins/registry.rb:137:in `lookup'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/plugins/registry.rb:179:in `lookup_pipeline_plugin'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/plugin.rb:140:in `lookup'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/pipeline.rb:114:in `plugin'", "(eval):8:in `initialize'", "org/jruby/RubyKernel.java:1079:in `eval'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/pipeline.rb:81:in `initialize'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/pipeline.rb:179:in `initialize'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/pipeline_action/create.rb:35:in `execute'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:340:in `converge_state'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:139:in `with_pipelines'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:337:in `converge_state'", "org/jruby/RubyArray.java:1613:in `each'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:324:in `converge_state'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:164:in `converge_state_and_update'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:139:in `with_pipelines'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:162:in `converge_state_and_update'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/agent.rb:88:in `execute'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/logstash-core/lib/logstash/runner.rb:302:in `execute'", "org/jruby/RubyProc.java:281:in `call'", "/Users/dedemorton/BuildTesting/6.0.0-alpha2_40867bdc/logstash-6.0.0-alpha2/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/task.rb:24:in `initialize'"]}
[2017-06-05T09:44:22,416][DEBUG][logstash.plugins.registry] Problems loading the plugin with {:type=>"input", :name=>"dead_letter_queue"}
[2017-06-05T09:44:22,420][ERROR][logstash.agent           ] Failed to execute action {:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"LogStash::PluginLoadingError", :message=>"Couldn't find any input plugin named 'dead_letter_queue'. Are you sure this is correct? Trying to load the dead_letter_queue input plugin resulted in this error: Problems loading the requested plugin named dead_letter_queue of type input. Error: NameError NameError"}
```

**Comment by talevy** [comment]
should be fixed by a new release following these changes: https://github.com/logstash-plugins/logstash-input-dead_letter_queue/pull/3

**Comment by robbavey** [comment]
@dedemorton Can this be tested with the latest build?

**Comment by dedemorton** [comment]
@robbavey Confirmed that this is fixed in the latest snapshot build, logstash-6.0.0-alpha3-SNAPSHOT.zip, so I'm closing this isssue.
